rllib/utils/test_utils.py

```diff
@@ -820,3 +820,76 @@ def should_check_eval(experiment):
     }
 
     return result
+
+
+def check_same_batch(batch1, batch2) -> None:
+    """Check if both batches are (almost) identical.
+
+    For MultiAgentBatches, the step count and individual policy's
+    SampleBatches are checked for identity. For SampleBatches, identity is
+    checked as the almost numerical key-value-pair identity between batches
+    with ray.rllib.utils.test_utils.check().
+
+    Args:
+        batch1: Batch to compare against batch2
+        batch2: Batch to compare against batch1
+    """
+    # Avoids circular import
+    from ray.rllib.policy.sample_batch import SampleBatch, MultiAgentBatch
+
+    assert type(batch1) == type(
+        batch2
+    ), "Input batches are of different " "types {} and {}".format(
+        str(type(batch1)), str(type(batch2))
+    )
+
+    def check_sample_batches(_batch1, _batch2, _policy_id=None):
+        # unroll IDs only have to fit if both batches have them
+        unroll_id_1 = _batch1.pop("unroll_id", None)
+        unroll_id_2 = _batch2.pop("unroll_id", None)
+        if unroll_id_1 is not None and unroll_id_2 is not None:
```

Comment:
what if 1 is None, the other is not?

Reply:
I say that batches should still be considered equal in that case. 
Consider the following:
```
a = MultiAgentBatch({"agent0": SampleBatch({(...)}
slices = a.timeslices(1)
b = MultiAgentBatch.concat_samples(slices)
```
I believe check_same_batch(a, b) should return True in this case even though b has unroll_ids and a has not.

What is your opinion? I see that this is not a beautiful solution but I struggle to come up with something that makes more sense.

Reply:
I think either is reasonable, depending on use case I guess.
no need to over-think, can you just add a comment about this right here?.